gloss/Graphics/Gloss/Internals/Interface/Backend/GLFW.hs

```diff
@@ -71,9 +70,17 @@ glfwStateInit
         , mouseWheelPos = 0
         , dirtyScreen   = True
         , display       = return ()
-        , idle          = return () }
+        , idle          = return ()
+        , optWinHdl     = Nothing }
```

Comment:
Previously this was un-initialized in the original struct. Again, modern ghc was complaining loudly about this.